File: ios/MobillDriversProtocol/GraffitiModel32.m
//
//  GraffitiModel32.m
//  CoolLED1248
//
//  Created by 君同 on 2023/4/25.
//  Copyright © 2023 Haley. All rights reserved.
//

#import "GraffitiModel32.h"

@implementation GraffitiModel32

- (instancetype)init
{
    if (self = [super init]) {
        self.coverTypeGraffiti = 1;
        self.startRowGraffiti = 0;
        self.startColGraffiti = 0;
        self.widthDataGraffiti = 96;
        self.heightDataGraffiti = 32;
        self.showModelGraffiti = 1;
        self.speedDataGraffiti = 8;
        self.stayTimeGraffiti = 2;
        
        self.dataGraffiti = [[NSArray alloc] init];
    }
    return self;
}


- (void)encodeWithCoder:(NSCoder *)coder {
    [coder encodeObject:self.dataGraffiti forKey:@"dataGraffiti"];
    
    [coder encodeInt:self.coverTypeGraffiti forKey:@"coverTypeGraffiti"];
    [coder encodeInt:self.startRowGraffiti forKey:@"startRowGraffiti"];
    [coder encodeInt:self.startColGraffiti forKey:@"startColGraffiti"];
    [coder encodeInt:self.widthDataGraffiti forKey:@"widthDataGraffiti"];
    [coder encodeInt:self.heightDataGraffiti forKey:@"heightDataGraffiti"];
    [coder encodeInt:self.showModelGraffiti forKey:@"showModelGraffiti"];
    [coder encodeInt:self.speedDataGraffiti forKey:@"speedDataGraffiti"];
    [coder encodeInt:self.stayTimeGraffiti forKey:@"stayTimeGraffiti"];
}

- (instancetype)initWithCoder:(NSCoder *)coder
{
    self = [super init];
    if (self) {
        //解档
        self.dataGraffiti = [coder decodeObjectForKey:@"dataGraffiti"];
        
        self.coverTypeGraffiti = [coder decodeIntForKey:@"coverTypeGraffiti"];
        self.startRowGraffiti = [coder decodeIntForKey:@"startRowGraffiti"];
        self.startColGraffiti = [coder decodeIntForKey:@"startColGraffiti"];
        self.widthDataGraffiti = [coder decodeIntForKey:@"widthDataGraffiti"];
        self.heightDataGraffiti = [coder decodeIntForKey:@"heightDataGraffiti"];
        self.showModelGraffiti = [coder decodeIntForKey:@"showModelGraffiti"];
        self.speedDataGraffiti = [coder decodeIntForKey:@"speedDataGraffiti"];
        self.stayTimeGraffiti = [coder decodeIntForKey:@"stayTimeGraffiti"];
    }
    return self;
}

@end

File: android/app/src/main/java/com/mobilldriversprotocol/models/GraffitiModel32.java
package com.mobilldriversprotocol.models;

import java.util.List;

/**
 * GraffitiModel32 - Graffiti/Image Content Data Format
 * Port from iOS GraffitiModel32.h
 * Copyright © 2024 Mobill. All rights reserved.
 */
public class GraffitiModel32 {

    // Graffiti pixel data - 2D array of color data
    private List<List<List<Float>>> dataGraffiti;

    // Content display mixing mode with other layers
    private int coverTypeGraffiti;

    // Content display starting row
    private int startRowGraffiti;

    // Content display starting column
    private int startColGraffiti;

    // Content display width
    private int widthDataGraffiti;

    // Content display height
    private int heightDataGraffiti;

    // Graffiti display mode (1=static, 2=left scroll, 3=right scroll, etc.)
    private int showModelGraffiti;

    // Graffiti display speed (speed for the display mode)
    private int speedDataGraffiti;

    // Graffiti stay time (pause time after one screen is displayed)
    private int stayTimeGraffiti;

    // Constructor
    public GraffitiModel32(int coverType, int startRow, int startCol, int width, int height) {
        this.coverTypeGraffiti = coverType;
        this.startRowGraffiti = startRow;
        this.startColGraffiti = startCol;
        this.widthDataGraffiti = width;
        this.heightDataGraffiti = height;
    }

    // Getters and Setters
    public List<List<List<Float>>> getDataGraffiti() {
        return dataGraffiti;
    }

    public void setDataGraffiti(List<List<List<Float>>> dataGraffiti) {
        this.dataGraffiti = dataGraffiti;
    }

    public int getCoverTypeGraffiti() {
        return coverTypeGraffiti;
    }

    public void setCoverTypeGraffiti(int coverTypeGraffiti) {
        this.coverTypeGraffiti = coverTypeGraffiti;
    }

    public int getStartRowGraffiti() {
        return startRowGraffiti;
    }

    public void setStartRowGraffiti(int startRowGraffiti) {
        this.startRowGraffiti = startRowGraffiti;
    }

    public int getStartColGraffiti() {
        return startColGraffiti;
    }

    public void setStartColGraffiti(int startColGraffiti) {
        this.startColGraffiti = startColGraffiti;
    }

    public int getWidthDataGraffiti() {
        return widthDataGraffiti;
    }

    public void setWidthDataGraffiti(int widthDataGraffiti) {
        this.widthDataGraffiti = widthDataGraffiti;
    }

    public int getHeightDataGraffiti() {
        return heightDataGraffiti;
    }

    public void setHeightDataGraffiti(int heightDataGraffiti) {
        this.heightDataGraffiti = heightDataGraffiti;
    }

    public int getShowModelGraffiti() {
        return showModelGraffiti;
    }

    public void setShowModelGraffiti(int showModelGraffiti) {
        this.showModelGraffiti = showModelGraffiti;
    }

    public int getSpeedDataGraffiti() {
        return speedDataGraffiti;
    }

    public void setSpeedDataGraffiti(int speedDataGraffiti) {
        this.speedDataGraffiti = speedDataGraffiti;
    }

    public int getStayTimeGraffiti() {
        return stayTimeGraffiti;
    }

    public void setStayTimeGraffiti(int stayTimeGraffiti) {
        this.stayTimeGraffiti = stayTimeGraffiti;
    }
}
